### PR TITLE
Slack user count extraction scheduler

### PR DESCRIPTION
**Pull request by venkuppu-chn**
### Changes
 - [ ] Why is this change required? What problem does it solve?
 - [ ] If proposing a new change then please raise an issue first

### How Has This Been Tested? (Optional)
 - [ ] Please describe in detail how you tested your changes.
 - [ ] Include details of your testing environment, and the tests you ran to
 - [ ] How your change affects other areas of the code, etc.

### Checklist
 - [ ] tested locally
 - [ ] added new dependencies
 - [ ] updated the docs
 - [ ] added a test

### CODE REVIEW

File: .github/workflows/slack.yml [hunk 1]
```diff
@@ -24,33 +24,42 @@ jobs:
...
+          # pip3 install python-dotenv
+
       # Runs a script using the runners shell
       - name: Run a user extraction script
-        working-directory: ./metrics/slack          
+        working-directory: ./metrics/slack
         env:
-          SLACK_KEY: ${{ secrets.SLACK_KEY }}
-          
+          MINIO: ${{ secrets.MINIO }}
```
Comment (shailesh-vaidya) [inline .github/workflows/slack.yml:38]: @venkuppu-chn  Just suggestion not sure if it's feasible. As I understand these are tokens/cookies for fetching data from Slack endpoints. Can we generate them runtime from user-name. We can take user-name as input and no need to store any secrets. 

@johnbent  Hope it's ok to use secrets with these names. Just confirming as we cleared Product-specific words from repositories as a part of contentious word removal activity.

File: .github/workflows/slack.yml [hunk 2]
```diff
@@ -24,33 +24,42 @@ jobs:
...
-      
-     # Push the slack pickle file to remote repo          
+          git config --local user.email "cortx.application@seagate.com"
+          git config --local user.name "cortx-admin"
+          git add ./metrics/pickles/slack_users_stats.pickle
+          git commit -m "Updating the slack user count"
+
+      # Push the slack pickle file to remote repo
       - name: Push changes
         uses: ad-m/github-push-action@master
```
Comment (mukul-seagate11) [inline .github/workflows/slack.yml:62]: what does @master means here? It is branch?

File: .github/workflows/slack.yml [hunk 3]
```diff
@@ -24,33 +24,42 @@ jobs:
...
-         git config --local user.email "cortx.application@seagate.com"
-         git config --local user.name "cortx-admin"
-         git add ./metrics/pickles/slack_users_stats.pickle
-         git commit -m "Update the slack user count"
-      
-     # Push the slack pickle file to remote repo          
+          git config --local user.email "cortx.application@seagate.com"
+          git config --local user.name "cortx-admin"
+          git add ./metrics/pickles/slack_users_stats.pickle
+          git commit -m "Updating the slack user count"
```
Comment (shailesh-vaidya) [inline .github/workflows/slack.yml:58]: @venkuppu-chn Can we add uniqueness to the commit message.  Maybe `$date` in the commit message.

File: .github/workflows/slack.yml [hunk 4]
```diff
@@ -24,33 +24,42 @@ jobs:
...
+
+      # Push the slack pickle file to remote repo
       - name: Push changes
         uses: ad-m/github-push-action@master
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
-          branch: ${{ github.ref }} 
+          branch: ${{ github.ref }}
```
Comment (shailesh-vaidya) [inline .github/workflows/slack.yml:65]: Can we hard code this to `main` branch.

File: .github/workflows/slack.yml (hunk 2 above)
Comment (venkuppu-chn) [reply .github/workflows/slack.yml:62]: @mukul-seagate11, I'm using Github action to push pickle file into appropriate branch. Yes, it is branch.

Comment (mukul-seagate11) [reply .github/workflows/slack.yml:62]: OK, than in that case, there will be concerns related to **contentious word** (i.e. master branch)

@johnbent, are we OK with this?

File: .github/workflows/slack.yml (hunk 1 above)
Comment (venkuppu-chn) [reply .github/workflows/slack.yml:38]: @shailesh-vaidya It is feasible. But, username & password authentication is deprecated due to security reasons not only in slack, so I suggest opting API based authentication.  Moreover, for the script to run, one has to join all the workspace to get metrices count. So, I suggest opting service account with their token to get information without having to join. Also, workflow is not designed for individuals.

File: .github/workflows/slack.yml (hunk 4 above)
Comment (venkuppu-chn) [reply .github/workflows/slack.yml:65]: @shailesh-vaidya , I have hard coded in line number 26. Kindly refer the below.
      - uses: actions/checkout@v2
        with:
          ref: main

File: .github/workflows/slack.yml (hunk 3 above)
Comment (venkuppu-chn) [reply .github/workflows/slack.yml:58]: @shailesh-vaidya, Sure.

File: .github/workflows/slack.yml (hunk 1 above)
Comment (shailesh-vaidya) [reply .github/workflows/slack.yml:38]: @venkuppu-chn  Thanks for the info. So can we use a single API token for all workspaces to reduce secrets count to one.  I agree we should use a service account for fetching info.

Comment (venkuppu-chn) [reply .github/workflows/slack.yml:38]: @shailesh-vaidya, I have verified it is not possible, we should require the dedicated token for each workspace.

File: .github/workflows/slack.yml (hunk 3 above)
Comment (venkuppu-chn) [reply .github/workflows/slack.yml]: I have added date in commit message

**Review by johnbent — APPROVED** [review]
Thanks @venkuppu-chn !

**Review by mukul-seagate11 — APPROVED** [review]
looks good

**Review by shailesh-vaidya — APPROVED** [review]

**Pull request merged**